tough/src/schema/de.rs

```diff
@@ -22,15 +22,18 @@ where
         key: Key,
         map: &mut HashMap<Decoded<Hex>, Key>,
     ) -> Result<(), error::Error> {
-        let calculated = key.key_id()?;
         let keyid_hex = hex::encode(&keyid);
-        ensure!(
-            keyid == calculated,
-            error::InvalidKeyIdSnafu {
-                keyid: &keyid_hex,
-                calculated: hex::encode(&calculated),
-            }
-        );
+        #[cfg(not(feature = "tap-12"))]
+        {
+            let calculated = key.key_id()?;
+            ensure!(
+                keyid == calculated,
+                error::InvalidKeyIdSnafu {
+                    keyid: &keyid_hex,
+                    calculated: hex::encode(&calculated),
+                }
+            );
+        }
```

Comment:
My reading of TAP-12 suggests that it's not sufficient to just remove this check, because two different key IDs can now refer to the same key:
> To ensure that a client does not use the same key to verify a signature more than once, they must additionally check that all keys applied to a signature threshold are unique. So, the specification should additionally require that "Clients MUST use each key only once during a given signature verification."

The existing code may not implement this requirement because before there was a 1:1 mapping between key IDs and keys, so the `ensure!` check here was sufficient.